cmd/subctl/recover_broker_info.go

```diff
@@ -0,0 +1,176 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subctl
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/subctl/internal/cli"
+	"github.com/submariner-io/subctl/internal/constants"
+	"github.com/submariner-io/subctl/internal/exit"
+	"github.com/submariner-io/subctl/internal/restconfig"
+	"github.com/submariner-io/subctl/pkg/broker"
+	"github.com/submariner-io/subctl/pkg/cluster"
+	"github.com/submariner-io/submariner-operator/api/v1alpha1"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var recoverRestConfigProducer = restconfig.NewProducer().WithPrefixedContext("broker").
+	WithDefaultNamespace(constants.DefaultBrokerNamespace).
+	WithPrefixedNamespace("broker", constants.DefaultBrokerNamespace)
+
+// recoverBrokerInfo represents the reconstruct command.
+var recoverBrokerInfo = &cobra.Command{
+	Use:   "recover-broker-info",
+	Short: "Recovers the broker-info.subm file from the installed Broker",
+	Run: func(cmd *cobra.Command, args []string) {
+		status := cli.NewReporter()
+		// if --brokerconfig flag provided, get the broker and proceed to get Submariner
+		contextFound, err := recoverRestConfigProducer.RunOnSelectedPrefixedContext("broker",
+			recoverBrokerFromConfigContext, status)
+		exit.OnError(err)
+
+		// if --brokerconfig not provided, search for broker on current context and proceed to get Submariner
+		if !contextFound && err == nil {
```

Comment:
nit: I guess `exit.OnError(err)` on L#51 will make this check here unneeded.

```suggestion
		if !contextFound {
```